### PR TITLE
Feature/add modal for course detail

### PR DESCRIPTION
**Pull request by jhang-jhe-wei**
# Show Event Detail in modal
## User Story
When users click the event on the timetable, they can see the modal which contains the event detail popup

## FYI
image

### CODE REVIEW

**Comment by EnriqueSubias** [comment]
Done!

Some notes:
 - Animation stopped working when I replaced `inset: '10% 20%'` by this:
 (Adjusted position to center when making modal size fit to content size)
 ```tsx
    height: 'min-content',
    width: '60%',
    minWidth: 'min-content',
    // position: 'fixed',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
```

 - There are still some scrollbars in certain Events

 - Frequency seems to be empty (We could hide it if it's unknown)

Course Detail Modal

**Review by EnriqueSubias — COMMENTED** [review]
Thanks for getting the modal template for me just to fill in the content 👍🏻

File: app/javascript/pages/CourseSimulator/TimeTable/Event.tsx [hunk 1]
```diff
@@ -36,25 +94,91 @@ const Event = (props: EventProps) => {
...
+            <p>Location & Class</p>
+            <p>Frequency</p>
+          </div>
+          <div className="space-y-2 w-2/3 py-8 ml-4">
+            <p>{calculateDay(event.wday) }</p>
+            <p className="gap-2 inline-block inline-block flex items-center fill-black">
+            <CalendarIcon/>
+              { calculateTimeStart(event) }
+            </p>
+            <p className="gap-2 inline-block inline-block flex items-center fill-black">
```
Comment (jhang-jhe-wei) [inline app/javascript/pages/CourseSimulator/TimeTable/Event.tsx]: there are two display properties in the same line(inline-block, flex)

File: app/javascript/pages/CourseSimulator/TimeTable/Event.tsx [hunk 2]
```diff
@@ -16,6 +21,31 @@ const calculateTimeDuration = (event: GroupEventState) => {
...
+  return `${sHour}:${sMinute}`
+}
+
+const calculateTimeEnd = (event: GroupEventState) => {
+  const eHour = Math.floor(event.endedMinuteAt/60);
+  const eMinute = (event.endedMinuteAt % 60).toString().padStart(2, '0');
+  return `${eHour}:${eMinute}`
+}
+
+const calculateDay = (wday: number) => {
```
Comment (jhang-jhe-wei) [inline app/javascript/pages/CourseSimulator/TimeTable/Event.tsx:36]: 0 for Sunday
1 ~ 6 for Monday ~ Saturday

File: app/javascript/pages/CourseSimulator/TimeTable/Event.tsx [hunk 3]
```diff
@@ -28,6 +58,34 @@ interface EventProps {
...
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    padding: '1.5rem',
+    borderStyle: 'solid',
+    borderRadius: '1rem',
+  },
+};
+
+const mq = window.matchMedia("(max-width: 768px)");
```
Comment (jhang-jhe-wei) [inline app/javascript/pages/CourseSimulator/TimeTable/Event.tsx]: I prefer using CSS implement RWD

**Comment by jhang-jhe-wei** [comment]
Merged

**Pull request merged**